web/src/settings_org.js

```diff
@@ -262,21 +264,51 @@ function set_msg_move_limit_setting(property_name) {
     enable_or_disable_related_message_move_time_limit_setting(property_name, disable_setting);
 }
 
-function message_delete_limit_setting_enabled(setting_value) {
+function message_delete_limit_setting_enabled() {
     // This function is used to check whether the time-limit setting
-    // should be enabled. The setting is disabled when delete_own_message_policy
-    // is set to 'admins only' as admins can delete messages irrespective of
-    // time limit.
-    if (setting_value === settings_config.common_message_policy_values.by_admins_only.code) {
+    // should be enabled. The setting is disabled when every user
+    // who is allowed to delete their own messages is also allowed
+    // to delete any message in the organization.
+    const realm_delete_own_message_policy = Number.parseInt(
+        $("#id_realm_delete_own_message_policy").val(),
+        10,
+    );
```

Comment:
This can be `realm.realm_delete_own_message_policy`?

Reply:
The reasoning for this code is that "message delete limit dropdown" is checked for disabling even after the other message deletion setting dropdowns are updated but not saved. So we would need the currently selected value.

Reply:
oh ya, I thought it is automatically saved. Makes sense!

Reply:
Thanks for pointing that out. I have updated it now.